src/plugins/es_ui_shared/public/components/code_editor/code_editor.test.tsx

```diff
@@ -0,0 +1,105 @@
+  test('is rendered', () => {
+    const component = mount(<EuiCodeEditor {...requiredProps} />);
+    expect(takeMountedSnapshot(component)).toMatchSnapshot();
+  });
+
+  describe('props', () => {
+    describe('isReadOnly', () => {
+      test('renders alternate hint text', () => {
+        const component = mount(<EuiCodeEditor isReadOnly />);
+        expect(takeMountedSnapshot(component)).toMatchSnapshot();
+      });
+    });
+
+    describe('theme', () => {
+      test('renders terminal theme', () => {
+        const component = mount(<EuiCodeEditor theme="terminal" />);
+        expect(takeMountedSnapshot(component)).toMatchSnapshot();
+      });
+    });
+
+    describe('aria attributes', () => {
+      test('allows setting aria-labelledby on textbox', () => {
+        const component = mount(<EuiCodeEditor aria-labelledby="labelledbyid" />);
+        expect(takeMountedSnapshot(component)).toMatchSnapshot();
+      });
+
+      test('allows setting aria-describedby on textbox', () => {
+        const component = mount(<EuiCodeEditor aria-describedby="describedbyid" />);
+        expect(takeMountedSnapshot(component)).toMatchSnapshot();
+      });
+    });
+  });
+
+  describe('behavior', () => {
+    let component: ReactWrapper;
+
+    beforeEach(() => {
+      component = mount(<EuiCodeEditor />);
+    });
+
+    describe('hint element', () => {
+      test('should be tabable', () => {
+        const hint = findTestSubject(component, 'codeEditorHint').getDOMNode();
+        expect(hint).toMatchSnapshot();
+      });
+
+      test('should be disabled when the ui ace box gains focus', () => {
+        const hint = findTestSubject(component, 'codeEditorHint');
+        hint.simulate('keyup', { key: keys.ENTER });
+        expect(findTestSubject(component, 'codeEditorHint').getDOMNode()).toMatchSnapshot();
+      });
+
+      test('should be enabled when the ui ace box loses focus', () => {
+        const hint = findTestSubject(component, 'codeEditorHint');
+        hint.simulate('keyup', { key: keys.ENTER });
+        // @ts-ignore onBlurAce is known to exist and its params are only passed through to the onBlur callback
+        component.instance().onBlurAce();
+        expect(findTestSubject(component, 'codeEditorHint').getDOMNode()).toMatchSnapshot();
+      });
+    });
+
+    describe('interaction', () => {
+      test('bluring the ace textbox should call a passed onBlur prop', () => {
+        const blurSpy = jest.fn().mockName('blurSpy');
+        const el = mount(<EuiCodeEditor onBlur={blurSpy} />);
+        // @ts-ignore onBlurAce is known to exist and its params are only passed through to the onBlur callback
+        el.instance().onBlurAce();
+        expect(blurSpy).toHaveBeenCalled();
+      });
+
+      // TODO: Fix this. document.activeElement resolves to be <body />, but the component behaves
+      // as expected in the browser.
+      test.skip('pressing escape in ace textbox will enable overlay', () => {
```

Comment:
@chandlerprall Do you or anyone on the EUI team have any suggestions as to why this test might be failing?

Reply:
Kibana's version of jsdom (16.4.0) is newer than EUI's version (11.12.0), and the newer one is preventing the call to `.focus()` from doing work because of this additional check: https://github.com/jsdom/jsdom/blob/16.4.0/lib/jsdom/living/helpers/focusing.js#L25-L27

```
    if (!elImpl.isConnected) {
      return false;
    }
```

Reply:
Thank you! This helped a ton.